sound/soc/sof/intel/hda-dsp.c

```diff
@@ -377,18 +388,16 @@ static int hda_resume(struct snd_sof_dev *sdev)
 			"error: failed to ready controller during resume\n");
 		return ret;
 	}
-#endif
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
-	/* turn off the links that were off before suspend */
-	list_for_each_entry(hlink, &bus->hlink_list, list) {
-		if (!hlink->ref_count)
-			snd_hdac_ext_bus_link_power_down(hlink);
-	}
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTCTL,
+				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN,
+				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN);
```

Comment:
I wonder if there is a disconnect here. It's fine to use the hdac library when SOF_HDA is defined. It's not ok when it's not. You don't have to replace all the code everywhere to get rid of hdac, only where we don't want it.

Reply:
@plbossart OK, changed, now should be easy to review. My first purpose it to rewrite a general part for both HDA and NOCODEC, and moved HDA part below together.